smtp-crawler/src/main/java/be/dnsbelgium/mercator/smtp/domain/crawler/SmtpAnalyzer.java

```diff
@@ -130,21 +126,24 @@ private SmtpVisitEntity doCrawl(String domainName) {
     }
   }
 
-  private Optional<List<SmtpHostEntity>> createSmtpHosts(String domainName, int priority) {
+  private Optional<List<SmtpHost>> createSmtpHosts(String domainName, int priority) {
     List<InetAddress> addresses = mxFinder.findIpAddresses(domainName);
     if (addresses.size() == 0) {
       logger.debug("No addresses found for {}", domainName);
       return Optional.empty();
     }
     logger.debug("We found {} addresses for {}", addresses.size(), domainName);
-    List<SmtpHostEntity> hosts = new ArrayList<>();
+    List<SmtpHost> hosts = new ArrayList<>();
     for (InetAddress address : addresses) {
-      SmtpHostEntity host = new SmtpHostEntity();
-      host.setHostName(domainName);
-      host.setPriority(priority);
+      SmtpHost host = new SmtpHost();
       SmtpConversation smtpConversation = crawl(address);
       smtpConversation.clean();
-      host.setConversation(smtpConversation);
+      if (smtpConversation.getId() != null){
+        host = host.fromCache(null, null, domainName, priority, new SmtpConversationEntity().fromSmtpConversation(smtpConversation));
```

Comment:
in terms of efficiency: we're now creating two instances of SmtpConversationEntity. It might be better to write this as `setFromSmtpConversation` and not create a second instance.